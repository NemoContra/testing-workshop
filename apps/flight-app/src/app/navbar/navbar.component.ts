import { Component } from '@angular/core';

@Component({
  selector: 'flight-navbar',
  templateUrl: 'navbar.component.html',
})
export class NavbarComponent {
  private sidebarVisible = false;

  sidebarToggle() {
    const body = document.getElementsByTagName('body')[0];

    if (!this.sidebarVisible) {
      body.classList.add('nav-open');
      this.sidebarVisible = true;
    } else {
      this.sidebarVisible = false;
      body.classList.remove('nav-open');
    }
  }
}
