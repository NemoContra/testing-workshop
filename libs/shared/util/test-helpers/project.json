{
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util/test-helpers/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/util/test-helpers/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared/util/test-helpers"],
      "options": {
        "jestConfig": "libs/shared/util/test-helpers/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["scope:shared", "type:util"]
}
