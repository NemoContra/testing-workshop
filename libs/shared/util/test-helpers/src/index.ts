export * from './lib/test-id-selector';
export * from './lib/create-spy-observer';
