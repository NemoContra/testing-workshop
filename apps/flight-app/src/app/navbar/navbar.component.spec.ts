import { NavbarComponent } from './navbar.component';
import { createHostFactory } from '@ngneat/spectator/jest';
import { testIdSelector } from '@testing-workshop/shared/util/test-helpers';

describe('NavbarComponent', () => {
  const createComponent = createHostFactory(NavbarComponent);

  const selectors = {
    navbarToggle: testIdSelector('navbar-toggle'),
  } as const;

  it('should match snapshot', () => {
    const spectator = createComponent(`<flight-navbar></flight-navbar>`);
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should open and close the sidebar', () => {
    const spectator = createComponent(`<flight-navbar></flight-navbar>`);

    // Toggle the sidebar open
    spectator.click(selectors.navbarToggle);
    spectator.detectChanges();
    const body = spectator.query('body', { root: true });
    expect(body?.classList).toMatchInlineSnapshot(`
      DOMTokenList {
        "0": "nav-open",
      }
    `);

    // Toggle the sidebar closed
    spectator.click(selectors.navbarToggle);
    spectator.detectChanges();
    expect(body?.classList).toMatchInlineSnapshot(`DOMTokenList {}`);
  });
});
