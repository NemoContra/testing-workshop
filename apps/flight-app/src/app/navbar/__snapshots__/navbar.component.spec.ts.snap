// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent should match snapshot 1`] = `
<ng-component
  __ngContext__="0"
>
  <flight-navbar>
    <nav
      class="navbar navbar-default"
    >
      <div
        class="container-fluid"
      >
        <div
          class="navbar-header"
        >
          <button
            class="navbar-toggle"
            data-tid="navbar-toggle"
            type="button"
          >
            <span
              class="sr-only"
            >
              Toggle navigation
            </span>
            <span
              class="icon-bar bar1"
            />
            <span
              class="icon-bar bar2"
            />
            <span
              class="icon-bar bar3"
            />
          </button>
          <a
            class="navbar-brand"
            href="#"
          >
            Flight42
          </a>
        </div>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="nav navbar-nav navbar-right"
          >
            <li>
              <a
                class="dropdown-toggle"
                data-toggle="dropdown"
                href="#"
              >
                <i
                  class="ti-panel"
                />
                <p>
                  Stats
                </p>
              </a>
            </li>
            <li
              class="dropdown"
            >
              <a
                class="dropdown-toggle"
                data-toggle="dropdown"
                href="#"
              >
                <i
                  class="ti-bell"
                />
                <p
                  class="notification"
                >
                  5
                </p>
                <p>
                  Notifications
                </p>
                <b
                  class="caret"
                />
              </a>
              <ul
                class="dropdown-menu"
              >
                <li>
                  <a
                    href="#"
                  >
                    Notification 1
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Notification 2
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Notification 3
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Notification 4
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Another notification
                  </a>
                </li>
              </ul>
            </li>
            <li>
              <a
                href="#"
              >
                <i
                  class="ti-settings"
                />
                <p>
                  Settings
                </p>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </flight-navbar>
</ng-component>
`;
